ui/app/common/displaycontrols/forms/controllers/versionedFormController.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+angular.module('bahmni.common.displaycontrol.forms')
+    .controller('versionedFormController', ['$scope', 'formService', 'appService', '$q', '$state',
+        function ($scope, formService, appService, $q, $state) {
+            $scope.shouldPromptBrowserReload = true;
+            $scope.showFormsDate = appService.getAppDescriptor().getConfigValue("showFormsDate");
+
+            const getUniqueForms = function (formData) {
+                const uniqueForms = [];
+                _.each(formData, function (item) {
+                    const foundElement = _.find(uniqueForms, function (filteredItem) {
+                        return item.formName === filteredItem.formName;
+                    });
+                    if (foundElement === undefined) {
+                        uniqueForms.push(item);
+                    }
+                });
+                return uniqueForms;
+            };
+
+            const sortFormDataByLatestDate = function (formData) {
+                return _.sortBy(formData, "encounterDateTime").reverse();
+            };
+
+            function filterForms (formData) {
+                const filteredForms = [];
+                _.each(formData, function (item) {
+                    const foundElement = _.includes($scope.section.formGroup, item.formName);
+                    if (foundElement) {
+                        filteredForms.push(item);
+                    }
+                });
+                return filteredForms;
+            }
+
+            var init = function () {
+                $scope.noFormFoundMessage = "No Form found for this patient";
```

Comment:
I think this message should be configurable and internationlized. Other than this I see no problems